tf5muxserver/mux_server_ListResource.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tf5muxserver
+
+import (
+	"context"
+	"slices"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+
+	"github.com/hashicorp/terraform-plugin-mux/internal/logging"
+)
+
+func (s *muxServer) ListResource(ctx context.Context, req *tfprotov5.ListResourceRequest) (*tfprotov5.ListResourceServerStream, error) {
+	rpc := "ListResource"
+	ctx = logging.InitContext(ctx)
+	ctx = logging.RpcContext(ctx, rpc)
+
+	server, diags, err := s.getListResourceServer(ctx, req.TypeName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// slices.all package
+
+	if diagnosticsHasError(diags) {
+		return &tfprotov5.ListResourceServerStream{
+			Results: slices.Values([]tfprotov5.ListResourceResult{
```

Comment:
✨ ✨ ✨ 

Want to add a comment to describe what's happening here? I think I'll do the same in Framework to explain the not-obvious agreement to "return a single result with an error diagnostic."